### ISSUE TITLE
assertion failure when binding UNSET as the set of removed elements in a non-frozen collection

### ISSUE THREAD

**Issue by kbr-** [scylladb/scylla#7825]
Suppose we have a table:
```
create table ks1.m (pk int primary key, v map<int, int>)
```

The following Python program works as expected:
```
from cassandra.cluster import Cluster
from cassandra.query import UNSET_VALUE

with Cluster() as cluster:
    session = cluster.connect()

    p = session.prepare('update ks1.m set v = :val where pk = 0');
    session.execute(p.bind({'val': {1:1, 2:2}}))
    session.execute(p.bind({'val': UNSET_VALUE}))
    session.execute(p.bind({}))
```

the second and third statement (which don't bind a value for `:val`) don't have an effect.

Similarily, this works as expected:
```
    p = session.prepare('update ks1.m set v = v + :val where pk = 0');
    session.execute(p.bind({'val': {3:3}}))
    session.execute(p.bind({'val': UNSET_VALUE}))
    session.execute(p.bind({}))
```

However, the following causes a crash:
```
    p = session.prepare('update ks1.m set v = v - :val where pk = 0');
    session.execute(p.bind({'val': {2:2}}))
    session.execute(p.bind({'val': UNSET_VALUE}))
    session.execute(p.bind({}))
```
either of the second or third statement causes a crash.

It's an assertion failure:
```
scylla: cql3/sets.cc:323: virtual void cql3::sets::discarder::execute(mutation &, const clustering_key_prefix &, const cql3::update_parameters &): Assertion `svalue' failed.
```

relevant code:
```
void
sets::discarder::execute(mutation& m, const clustering_key_prefix& row_key, const update_parameters& params) {
    assert(column.type->is_multi_cell()); // "Attempted to remove items from a frozen set";

    auto&& value = _t->bind(params._options);
    if (!value) {
        return;
    }

    collection_mutation_description mut;
    auto kill = [&] (bytes idx) {
        mut.cells.push_back({std::move(idx), params.make_dead_cell()});
    };
    auto svalue = dynamic_pointer_cast<sets::value>(value);
    assert(svalue);
    mut.cells.reserve(svalue->_elements.size());
    for (auto&& e : svalue->_elements) {
        kill(e);
    }
    m.set_cell(row_key, column, mut.serialize(*column.type));
}
```

**Comment by nyh** [comment]
I think this is a duplicate for #7740 which I opened recently, and have a detailed reproducing test for (if it's not a duplicate, then it's closely related).
@dekimir posted a patch for the assertion failure - see https://github.com/scylladb/scylla/pull/7870 - but it only avoids the assertion failure, it doesn't actually do the right thing (although maybe for deletion, it will be good enough). Can you please check if this patch solves your problem?

**Comment by nyh** [comment]
@kbr- can we close this issue? Was your problem fixed by the fix for #7740, or does a problem remain?
